developer_setup.md

```diff
@@ -1,323 +1,213 @@
-* There is also a minimal mode available to start the application with fewer services and workers for faster startup or
-targeted end-user testing. See the [minimal mode guide](developer_setup/minimal_mode.md) for details.
-* As an alternative to minimal mode, individual workers can also be started using [Foreman](https://ddollar.github.io/foreman/)
-  * See the [Foreman guide](developer_setup/foreman.md) for details.
-* To run the test suites, see [the guide on that topic](developer_setup/running_test_suites.md).
+---
 
-### Update dependencies and migrate db
+Configure the cluster.
 
-* You can update ruby and javascript dependencies as well as run migrations using one command
+On Fedora and CentOS, configure a cluster using `postgresql-setup`.
 
 ```bash
-bin/update                # Updates dependencies using bundler and bower, runs migrations, prepares test db.
+sudo PGSETUP_INITDB_OPTIONS='--auth trust --username root --encoding UTF-8 --locale C' postgresql-setup --initdb
 ```
 
-For provider, UI or other plugin development, see [the guide on that topic](developer_setup/plugins.md).
+On Debian and Ubuntu, configure a cluster using `pg_createcluster`.
 
-#### Some troubleshooting notes
+```bash
+sudo pg_dropcluster --stop 10 main
+sudo pg_createcluster -e UTF-8 -l C 10 main -- --auth trust --username root
+```
 
-* First login fails
+On macOS, configured a cluster using `initdb` directly.
 
-Make sure you have memcached running. If not stop the server with `bundle exec rake evm:stop`,
-start memcached and retry.
+```bash
+rm -rf /usr/local/var/postgres
+initdb --auth trust --username root --encoding UTF-8 --locale C /usr/local/var/postgres
+```
 
-* `bin/setup fails` while trying to load the gem 'sprockets/es6'
+Enable and start the service.
 
-If this happens check the log for
-`ExecJS::RuntimeUnavailable: Could not find a JavaScript runtime` a few lines down.
-When this message is present, then the you need to install `node.js` and re-try
+Using systemd:
 
-* `bin/setup` fails while trying to install the 'nokogiri' gem
+```bash
+systemctl enable --now postgresql
+```
 
-If this happens, you may be missing developer tools in your OS X. Try to install them with
-`xcode-select --install` and re-try.
+Using Homebrew:
 
-* `bin/setup` fails to install the 'sys-proctable' gem, or installs the wrong version.
+```bash
+brew services start postgresql
+```
 
-If this happens it may be a Bundler issue. Try running `bundle config specific_platform true`
-and re-try.
+## Install nvm and JavaScript build utilities
 
-* Can't install any gems during bundle install
+Yarn, Gulp and Webpack are required to compile JavaScript assets. NodeJS version 10.16.x is required. If your distribution doesn't ship NodeJS 10.x, you can install [nvm](https://github.com/nvm-sh/nvm) and follow the setup steps. Then install `yarn`, `gulp-cli` and `webpack`.
 
+```bash
+nvm install 10
+npm install -g yarn gulp-cli webpack
```

Comment:
This fails for me without sudo...
```
 npm install -g yarn gulp-cli webpack
npm WARN checkPermissions Missing write access to /usr/lib/node_modules
npm ERR! path /usr/lib/node_modules
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall access
npm ERR! Error: EACCES: permission denied, access '/usr/lib/node_modules'
npm ERR!  { [Error: EACCES: permission denied, access '/usr/lib/node_modules']
npm ERR!   stack:
npm ERR!    'Error: EACCES: permission denied, access \'/usr/lib/node_modules\'',
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'access',
npm ERR!   path: '/usr/lib/node_modules' }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator (though this is not recommended).

npm ERR! A complete log of this run can be found in:
```

Reply:
Oh, that means you're still using the system `npm`. You need to restart your shell to source in the `nvm` shims (just like `rbenv`). I'll add a note about that.